File: golang/daily21/m01/d08.go
package y2021m01

// 每日一题 20210108
//
// 189. 旋转数组
// Link: https://leetcode-cn.com/problems/rotate-array/
func rotate(nums []int, k int) {
	newNums := make([]int, len(nums))
	for i, v := range nums {
		newNums[(i+k)%len(nums)] = v
	}
	copy(nums, newNums)
}

File: typescript/daily21/m07/d08.ts
import { runAllTest } from "../../utils/test"
/**
 * 930. 和相同的二元子数组
 * 
 * @link https://leetcode-cn.com/problems/binary-subarrays-with-sum/
 */
function numSubarraysWithSum(nums: number[], goal: number): number {
  let sum = 0;
  let cnt = new Map();
  let ret = 0;
  for (let num of nums) {
    cnt.set(sum, (cnt.get(sum) || 0) + 1);
    sum += num;
    ret += cnt.get(sum - goal) || 0;
  }
  return ret;
};

runAllTest("20210708", numSubarraysWithSum, [
  { args: [[1, 0, 1, 0, 1], 2], res: 4 },
  { args: [[0, 0, 0, 0, 0], 0], res: 15 },
]);